File: all-submissions/1666-make-the-string-great/solution.java
class Solution {
    public String makeGood(String s) {
        Stack<Character> st = new Stack<>();
        for (char c: s.toCharArray()) { 
            if(!st.empty() && Math.abs(st.peek() - c) == 32) {
                st.pop();
            }
            else {
                st.push(c);
            }
        }
        
        StringBuilder sb = new StringBuilder();
        while(!st.empty()) {
            sb.append(st.pop());
        }
        if (sb != null) { 
            sb.reverse();
        }
        return sb.toString();
        
    }
}

File: all-submissions/1666-make-the-string-great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        end = 0
        st = list(s)
        for i in range(len(st)):
            if (end > 0 and abs(ord(st[i]) - ord(st[end-1])) == 32):
                end -=1
            else:
                st[end] = st[i]
                end +=1
        return "".join(st[0:end])
